src/app/dashboard/learning-object-editor/learning-object-editor.component.ts

```diff
@@ -139,13 +138,48 @@ export class LearningObjectEditorComponent implements OnInit, OnDestroy {
   }
 
   save() {
-    this.learningObjectService.update(this.learningObject)
+    let contributorIds: string[] = [];
+    if (this.learningObject.contributors && this.learningObject.contributors.length) {
+      contributorIds = this.learningObject.contributors.map(user => user.id);
+    }
+    const updateObject = {
+      id: this.learningObject.id,
+      name: this.learningObject.name,
+      length: this.learningObject.length,
+      description: this.learningObject.description,
+      levels: this.learningObject.levels,
+      contributors: contributorIds
+    }
+
+    this.learningObjectService.update(updateObject)
+      .then(
+        () => {
+          this.open(true);
+        },
+        error => {
+          this.open(false);  
+        }
+      );
+  }
+
+  saveOutcome(index: number) {
+
+    const learningObjectId = this.learningObject.id;
+    const learningOutcomeId = this.learningObject.outcomes[index].id;
+    
+    const updateOutcome = {
+      bloom: this.learningObject.outcomes[index].bloom,
+      verb: this.learningObject.outcomes[index].verb,
+      text: this.learningObject.outcomes[index].text,
+    }
+
+    this.learningObjectService.updateOutcome(updateOutcome, learningObjectId, learningOutcomeId)
       .then(
         () => {
           this.open(true);
         },
         error => {
-          this.open(false);
+          this.open(false);  
```

Comment:
Extra spaces were added:

```suggestion
          this.open(false);
```